Stidents.BLL/Services/StudentService.cs

```diff
@@ -166,17 +167,18 @@ public async Task<int> GetCount(string searchString, EnumSearchParameters search
         {
             if (string.IsNullOrEmpty(searchString) || searchParametr == EnumSearchParameters.None)
             {
-                return (await _unitOfWork.StudentRepository.GetStudentListAsync()).Count();
+                var spec = new StudentWithItemsSpecifications();
+                return (await _unitOfWork.StudentRepository.CountAsync(spec));
             }
             return (await SearchAllAsync(searchString, searchParametr)).Count();
         }
-
+        #region GetPaginatedResult
         public async Task<IEnumerable<Student>> GetPaginatedResult(int currentPage, int pageSize = 10)
         {
             return (await _unitOfWork.StudentRepository.GetStudentListAsync())
```

Comment:
It will get all students from DB and then Take only needed page? It's not optimal